src/lib/snapp.ts

```diff
@@ -372,6 +373,7 @@ export class SmartContract {
   static compile(address?: PublicKey) {
     // TODO: think about how address should be passed in
     // if address is not provided, create a random one
+    // TODO: maybe PublicKey should just become a variable? Then compile doesn't need to know the address, which seems more natural
```

Comment:
👍 -- calling @mrmr1993 for a double-check

Reply:
Yea, make it a variable

Reply:
Will do that, thanks! Outsourced it to a separate issue, since it will be easier after other refactorings I want to do: https://github.com/o1-labs/snarkyjs/issues/136